gossip/tx_scrambler.go

```diff
@@ -15,8 +19,46 @@ type ScramblerEntry interface {
 	Sender() common.Address
 	// Nonce returns the transaction nonce
 	Nonce() uint64
-	// GasPrice returns the transaction gasPrice price
-	GasPrice() uint64
+	// GasPrice returns the transaction gas price
+	GasPrice() *big.Int
+	// Unwrap returns the wrapped transaction.
+	Unwrap() *types.Transaction
```

Comment:
This is not a good interface design. In most contexts where you are using the `ScrambleEntry` interface right now you do not need the `Unwrap` function -- this is even more evident that one of the implementations of this function states "//unused".

Instead of just adding this method for convenience to an existing interface you should stay true to the intention of the interface. The `ScrambleEntry` is intended to provide an abstraction for everything that can be scrambled. If the function `FilterAndOrderTransactions` needs something more, create a new interface that extends this one.

Reply:
I agree this was not a good design. I removed the `Unwrap()` func and casted the interface back to the transactions.
My thoughts were that this file needs quite an extensive refactoring anyway, so casting is the easiest we can do rn and IMO its quite well readable.

When we start to refactor this, we might want to consider wrap the `types.Transactions` into an interface which actually implements `ScramblerEntry` so that we avoid some unnecessary looping.
Wdyt?

Reply:
You have just developed this file and it already needs extensive refactoring? Why not do it right now, before the integration?

Reply:
I meant that we might refactor `gossip/c_block_callbacks.go` even more, or do you reckon the changes in #266 should be enough?